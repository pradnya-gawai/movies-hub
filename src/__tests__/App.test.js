import { act, render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';
import App from '../App';

test('app rendering/navigating properly for all routes', async () => {
  const history = createMemoryHistory();
  act(() => {
    render(
      <Router history={history}>
        <App />
      </Router>
    );
  });

  // verify page content for expected route
  // often you'd use a data-testid or role query, but this is also possible
  expect(await screen.findByText(/trending today/i)).toBeInTheDocument();

  act(() => {
    userEvent.click(screen.getByTestId('movie-icon'));
  });
  expect(await screen.findByText(/discover movies/i)).toBeInTheDocument();

  act(() => {
    userEvent.click(screen.getByTestId('tv-icon'));
  });
  expect(await screen.findByText(/discover tv series/i)).toBeInTheDocument();

  act(() => {
    userEvent.click(screen.getByTestId('search-icon'));
  });
  expect(await screen.findByLabelText('Search')).toBeInTheDocument();
});
