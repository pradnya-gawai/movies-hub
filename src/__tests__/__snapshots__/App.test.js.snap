// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should test snapshot of app 1`] = `
<BrowserRouter>
  <Header />
  <div
    className="app"
    data-testid="app"
  >
    <WithStyles(ForwardRef(Container))>
      <Switch>
        <Route
          component={[Function]}
          exact={true}
          path="/"
        />
        <Route
          component={[Function]}
          path="/movies"
        />
        <Route
          component={[Function]}
          path="/series"
        />
        <Route
          component={[Function]}
          path="/search"
        />
      </Switch>
    </WithStyles(ForwardRef(Container))>
  </div>
  <MainNav />
</BrowserRouter>
`;
